packages/api/db/migration/20240504185218_create_soil_amendement_task_products_table.js

```diff
@@ -238,11 +239,12 @@ export const up = async function (knex) {
     table.integer('purpose_id').references('id').inTable('soil_amendment_purpose').notNullable();
     table.primary(['task_products_id', 'purpose_id']);
     table.string('other_purpose');
-    table.check(
-      '(?? IS NOT NULL AND ?? = ??) OR (?? IS NULL)',
-      ['other_purpose', 'purpose_id', otherPurpose.id, 'other_purpose'],
-      'other_purpose_id_check',
-    );
+    // LF-4246 - add back in this check
```

Comment:
You want to comment it? I think there's a chance the outcome of the LF-4246 investigation will be that we don't really care to do more migration on this table. Clean delete would let us go either way. This kinda feels like it locks us into another migration 😂

Reply:
Yeah I think it is helpful context. 🤷

Reply:
Ok. My preference not that strong 🙂